extensions/grunt/src/main.ts

```diff
@@ -101,6 +116,20 @@ class FolderDetector {
 		return this.promise;
 	}
 
+	public async getTask(_task: vscode.Task): Promise<vscode.Task | undefined> {
+		const gruntTask = (<any>_task.definition).task;
+		if (gruntTask) {
+			let kind: GruntTaskDefinition = {
+				type: 'grunt',
+				task: gruntTask
+			};
+			let options: vscode.ShellExecutionOptions = { cwd: this.workspaceFolder.uri.fsPath };
+			let task = new vscode.Task(kind, this.workspaceFolder, gruntTask, 'grunt', new vscode.ShellExecution(await this._gruntCommand, [gruntTask], options));
```

Comment:
For the ShellExecution, it should be done like lines 189-191 and inlcude the quoting for the space if needed.